resource/geo-density-chart.js

```diff
@@ -0,0 +1,379 @@
+            }
+            return x1 >= xmax || y1 >= ymax || x2 < xmin || y2 < ymin;
+        });
+        console.log("Results:", results);
+        return results;
+    };
+
+    /**
+     * Compute the closest point(s) with respect to the type of neighborhood expected
+     * @param lat: Center point latitude
+     * @param long: Center point longitude
+     * @param mode:
+     *      1 - Closest point from cursor (None if no point at all)
+     *      2 - The set of points in the rectangular search window with fixed search radius (None if no points in search window)
+     * @returns {*[]|(*|number|bigint)[]}
+     */
+    this.getNeighbors = function(lat, long, mode=1){
+        // If no quadtree defined or quadtree is empty return no neighbors
+        if (!_quadtree || _quadtree.size() === 0){
+            return [];
+        }
+        if (mode === 1){
+            console.log('Quadtree results', _quadtree.find(lat, long, 10));
+            return [_quadtree.find(lat, long, 10)]
+        } else if (mode === 2){
+            console.log("Searching for the points with getNeighbors ...");
+            let searchRadius = 0.001;
+            return this.search(_quadtree, lat-searchRadius, long-searchRadius,
+                lat+searchRadius, long+searchRadius)
+        }
+    };
+
+    /**
+     * Display the neighborhood on svg overlay
+     */
+    this.displayLocal = function(){
+
+        console.log("Call to displayLocal ...");
+        d3.selectAll("circle").remove();
+        // Define the d3 dynamics around the geospatial data points (rendering and events)
+        _svgPointer.selectAll("myCircles")
+            .attr("pointer-events", "visible")
+            .data(_closestMarker)
+            .enter()
+            .append("circle")
+            .attr("cx", function(d) {
+                _tooltip.html("<div>Lon: <strong>" + d.long + "</strong><br>Lat: <strong>"+d.lat+"</strong>" +
+                    formatTooltip(d.tooltip) + "</div>");
+                return _mapPointer.latLngToLayerPoint([d.lat, d.long]).x
+            })
+            .attr("cy", function(d) {
+                return _mapPointer.latLngToLayerPoint([d.lat, d.long]).y
+            })
+            .attr("id", "circleBasicTooltip")
+            .on('mouseover', function() { // handle the event user mouse is over the circle data point
+                _tooltip.style("visibility", "visible")
+                    .style("left", (d3.event.pageX+15) + "px")
+                    .style("top", (d3.event.pageY-25) + "px");
+                d3.select(this).transition()
+                    .duration('50')
+                    .attr("fill", "red")
+                    .attr('r', 10)
+            })
+            .on('mouseout', function() { // handle
+                _tooltip.style("visibility", "hidden");
+                d3.select(this).transition()
+                    .duration('150')
+                    .attr("r", 5)
+            })
```

Comment:
Indeed the tooltip is sometimes stuck and the link with the mouse is broken, don't know why and as you said, when the mouse event is detected again the tooltip follows showing expected behaviour. I will try to understand why but it might be difficult to reproduce

Reply:
**After introduction of the leaflet popup, the mouse outside the map leads to popup of the closest marker even if mouse is not over it, will need to investigate a bit**

Reply:
Fixed